x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/utils/InferenceProcessorInfoExtractor.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.utils;
+
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.metadata.Metadata;
+import org.elasticsearch.ingest.ConfigurationUtils;
+import org.elasticsearch.ingest.IngestMetadata;
+import org.elasticsearch.ingest.Pipeline;
+import org.elasticsearch.transport.Transports;
+
+import java.util.HashSet;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Consumer;
+
+import static org.elasticsearch.inference.InferenceResults.MODEL_ID_RESULTS_FIELD;
+import static org.elasticsearch.ingest.Pipeline.PROCESSORS_KEY;
+
+/**
+ * Utilities for extracting information around inference processors from IngestMetadata
+ * <p>
+ * this class was duplicated from org.elasticsearch.xpack.ml.utils.InferenceProcessorInfoExtractor
+ */
+
+public final class InferenceProcessorInfoExtractor {
```

Comment:
I couldn't figure out how to import this function from the ML module without causing Jarr hell, so I just duplicated it here.

Reply:
I tried moving `org.elasticsearch.xpack.ml.utils.InferenceProcessorInfoExtractor` to `package org.elasticsearch.xpack.core.ml.utils` and it seems like I can reference it from both the inference plugin and ml. I didn't replace all the ml plugin's references though. I did have to hardcode the `TYPE` like you mentioned below (maybe that could be moved too, I didn't look too hard).

Does moving it there cause issues for you?

Reply:
I did try that initially and it caused Jar hell, so I am hesitant to try that again. It seems like it should work, but for some reason it didn't work when I tried it.